presto-main/src/test/java/com/facebook/presto/type/TestTimestamp.java

```diff
@@ -237,6 +243,49 @@ public void testCastFromSlice()
     }
 
     @Test
+    public void testCastFromSliceFailsOnTimestampWithTimeZone()
+    {
+        if (connectorSession().isLegacyTimestamp()) {
+            // we expect those not to fail in legacy mode
+            return;
+        }
+
+        functionAssertions.assertFunctionThrowsPrestoException(
+                "cast('2001-1-22 03:04:05.321 +07:09' as timestamp)",
+                INVALID_CAST_ARGUMENT,
+                "Value cannot be cast to timestamp: 2001-1-22 03:04:05.321 +07:09");
```

Comment:
Hm. I'm not very happy about the implementation but this is how it looks f5a815d1ebc337f0c86699acf17e6089fc5a0207. I think it's nice to have though, as it is the case that worked before.